### ISSUE TITLE
Freeze dataset during review

### ISSUE THREAD

**Issue by posixeleni** [IQSS/dataverse#3172]
During the review period of the _Private URL for draft dataset peer review_ one stakeholder commented that the following functionality would also be helpful:

> My only concern is that if authors were to use this, they would be able to edit the data during peer review.  Which could mean that the peer reviewers were not all reviewing the same data, but that we would have no way of knowing this.

This suggestion is therefore for an option to freeze a dataset to allow others to review it. This can either be done by having version control at the draft level or providing some UI functionality that a user can choose to freeze a dataset until peer review has completed.

@scolapasta also mentioned that we already technically have the ability to freeze a dataset when someone is ingesting tabular files.

**Comment by pameyer** [comment]
I'd agree with the suggestion to freeze datasets when the dataset is being reviewed, either by an internal curator role or external reviewer via private URL.

**Comment by pdurbin** [comment]
I'm just adding the work "locked" to this issue to help me find it again since that's the term (rather than "freeze") that's been used elsewhere, most recently at http://irclog.iq.harvard.edu/dataverse/2017-01-10#i_47026

**Comment by pameyer** [comment]
And for the sake of not having information only in IRC logs, some additional details from that:

SUBMITTED -> REVIEWED -> PUBLISHED, and SUBMITTED - ( send back to user with feedback ) -> DRAFT; with SUBMITTED uneditable

**Comment by mheppler** [comment]
@pameyer +1 for salvaging useful info out of IRC and putting it to use in the GitHub issue. Thank you, sir.

**Comment by scolapasta** [comment]
@pameyer In the above, REVIEWED should also be uneditable, no?

Should we add the ability to lock / unlock any dataset, in general? with the review functionality calling it by default. (so sending would lock it, but it could be manually unlocked, if needed)

(this has been discussed in person some, but adding here for discussion)

**Comment by pameyer** [comment]
@scolapasta - thanks for catching that; REVIEWED should also be uneditable.

It seems to me like it makes sense to have generic lock/unlock, for use here and other places (although I don't have a list of other places).

**Comment by pdurbin** [comment]
@pameyer I just added the "SBGrid" label to this issue because I'm pretty sure this is a feature you want. Please check out related issue #3702 that was recently opened by @amberleahey

**Comment by pdurbin** [comment]
Can we close this issue now that #4139 is in the current sprint?

**Comment by pameyer** [comment]
The state transitions ("SUBMITTED -> REVIEWED -> PUBLISHED, and SUBMITTED - [ send back to user with feedback ] -> DRAFT; with SUBMITTED uneditable") from this issue aren't covered in 4139 (aka - send back with feedback and "reviewed" state are distinct from curators being able to edit).

I'm not sure if there's a current issue for feedback from the reviewer, or having a separate "reviewed" state (from what I remember of those discussions it's not something that's in the plan anywhere).

**Comment by pdurbin** [comment]
@pameyer is what you want captured in the "Submit for Review workflow" doc that @dlmurphy made? Here's a link: https://docs.google.com/a/harvard.edu/document/d/1UQmdu0SNJxKZSIKl7DNEaKROhQUfM5MKtfE3LuSpAk4/edit?usp=sharing

I can't tell if this issue is being tracked at https://trello.com/c/BriFJ5mR/35-submit-for-review-return-to-author-messages-3702-3943 or not. It seems highly related.

**Comment by dlmurphy** [comment]
That workflow doc isn't like a blueprint, it's describing the feature the way it is _now._ If my understanding is correct, I think the separate "reviewed" state that @pameyer is asking for would occur between step 5 and 6 - instead of directly publishing the dataset, the curator would signal that it has been reviewed and approved, and then it would take an additional step to publish it.

**Comment by pdurbin** [comment]
@dlmurphy ok. This is making more sense. Thanks. Do you feel like this issue is capturing that need? I don't.

@pameyer would you be willing to create a new issue about this? This old issue is really hard to follow. Or maybe @dlmurphy can create it? Then we can close this old confusing issue.

**Comment by pameyer** [comment]
@pdurbin that trello link is tracking the "send back with user feedback" part

**Comment by pdurbin** [comment]
@pameyer should this issue be closed? I think pull request #3992 (lock modes added) was the one where we implemented your requirement of preventing users from editing while the dataset is in review, but I could be mistaken.

**Comment by pameyer** [comment]
@pdurbin Looks that way (we don't have the full state diagram discussed here; but dataset authors not changing things during dataset review should be working).

**Comment by pdurbin** [comment]
@pameyer thanks! Pull request #3992 was merged as part of 4.8 so I gave this issue that milestone. Case closed! 😄
